src/extension.ts

```diff
@@ -51,23 +23,25 @@ export function activate(context: vscode.ExtensionContext): void {
     // ...dynamically create LSP clients
     ...config.languages.map(makeClient),
     // ...and dynamically create completions, definitions, and code actions providers.
-    ...config.languages.map(makeTags)
+    ...config.languages.map(makeTags),
+    // restart extension when config changes
+    vscode.workspace.onDidChangeConfiguration(ev => {
+      if (ev.affectsConfiguration(alloglot.config.root)) restart(context)
+    }),
+    // user command to restart extension
+    vscode.commands.registerCommand(alloglot.commands.restart, () => restart(context)),
   )
 }
 
-function getWorkspaceConfig(workspaceConfigPath: string): LanguageConfig[] {
-  const workspaceFolders =  vscode.workspace.workspaceFolders?.map(folder => folder.uri)
-    try {
-      if (workspaceFolders && workspaceFolders.length > 0)
-      {
-        const fullPath = vscode.Uri.joinPath(workspaceFolders[0], workspaceConfigPath);
-        return JSON.parse(readFileSync(fullPath.path,'utf-8')).languages;
-      } else {
-        return []
-      }
-    } catch (err){
-      return []
-    }
+export function deactivate() {
+  globalContext && disposeAll(globalContext)
+}
+
+function disposeAll(context: vscode.ExtensionContext) {
```

Comment:
Anyway we can make this kill the spawned processes here? So we don't need to do that trap stuff?

Reply:
VS Code does kill the spawned language server. Our problem is slightly different, though. Our "language server" is a script that forks `ghciwatch` and starts the real language server, `static-ls`. VS Code will kill `static-ls`. The `trap` causes `ghciwatch` to die whenever `static-ls` dies. Without `trap`, VS Code will kill `static-ls`, but `ghciwatch` will be left a zombie.